File: Javascript/섹션2. (진행중) 1-2차원 배열 탐색/2. 보이는 학생 (중하)/AA.js
function solution(num, a, b){
    let answer=[]
    
    for(let i=0;i<num;i++){
        if(a[i]==b[i]){
            console.log('D')
        }
        else if(a[i]==1){
            if(b[i]==2){
                console.log('B')
            }
            else if(b[i]==3){
                console.log('A')
            }
        }
        else if(a[i]==2){
            if(b[i]==1){
                console.log('A')
            }
            else if(b[i]==3){
                console.log('B')
            }
        }
        else if(a[i]==3){
            if(b[i]==1){
                console.log('B')
            }
            if(b[i]==2){
                console.log('A')
            }
        }
    }

    return answer
}
let a=[2, 3, 3, 1, 3];
let b=[1, 1, 2, 2, 3];
// arr=str.split(' ')
//console.log(solution(5, a, b))
solution(5, a, b)

File: Python/섹션 3 - (진행중) 탐색&시뮬레이션(string,1차원,2차원 리스트 탐색)/7. 사과나무 (상)/AA.py
import sys
# currentPath='C:\\_Algorithm-Study-Folder\\Algorithm-Study\\Python\\섹션 3 - (진행중) 탐색&시뮬레이션(string,1차원,2차원 리스트 탐색)\\7. 사과나무'
# sys.stdin=open(currentPath+'\\input.txt', 'rt')

n=int(input())
arr=[list(map(int,input().split())) for _ in range(n)]
# arr=[list(map(int,input().split()))] * num

rFlag=False
cFlag=False
sum=0
repeat=1
middle=int(n/2) #배열은 0부터이므로, -1 안 해줘도 됨
cursor=middle
for i in range(n):
    nowLoc=cursor
    for _ in range(repeat):
        sum+=arr[i][nowLoc]
        nowLoc+=1
        
    if repeat==n:
        rFlag=True
    if(rFlag):
        repeat-=2
        if repeat<0:
            break
    else:
        repeat+=2

    if cursor==0:
        cFlag=True
    if(cFlag):
        cursor+=1
    else:
        cursor-=1
print(sum)